includes/class-wcpdf-main.php

```diff
@@ -444,8 +444,15 @@ public function generate_document_ajax() {
 				}
 				break;
 			case 'full':
-				// no order
-				if ( ! $order ) {
+				// check the user privileges
+				$full_permission = WPO_WCPDF()->admin->user_can_manage_document( $document_type );
+
+				if ( $full_permission ) {
```

Comment:
I think you should follow the same procedure as the other checks, by setting the `$allowed = false;`.